src/theme/components/hero.ts

```diff
@@ -138,16 +196,21 @@ const primary = definePartsStyle({
     bodyText: {
       marginBottom: "0",
     },
-    // h1: { color: "ui.typography.inverse.heading" },
+    ".chakra-heading": {
+      color: getTextColor("heading", "light", foregroundColor, isDarkText),
+    },
     _dark: {
       bgColor: "dark.ui.bg.default",
-      color: "dark.ui.typography.body",
+      color: getTextColor("body", "dark", foregroundColor, isDarkText),
+      ".chakra-heading": {
+        color: getTextColor("heading", "dark", foregroundColor, isDarkText),
+      },
     },
   },
   heading: {
     color: "dark.ui.typography.heading",
```

Comment:
Should this also use the `getTextColor` function?

Reply:
Yes.